File: .opencode/skills/zmx-long-lived-shells/scripts/zmx_session.js
#!/usr/bin/env node
'use strict';

const { spawnSync } = require('child_process');

function runZmx(args) {
  const result = spawnSync('zmx', args, { stdio: 'inherit' });
  if (result.error && result.error.code === 'ENOENT') {
    console.error('zmx not found. Install with: brew install neurosnap/tap/zmx');
    return 127;
  }
  if (result.status === null || result.status === undefined) {
    return 1;
  }
  return result.status;
}

function usage() {
  console.error('Usage: zmx_session.js <attach|run|list|history|kill|detach|version> <session?> [-- command...]');
  return 2;
}

const argv = process.argv.slice(2);
const command = argv[0];
const session = argv[1];
const rest = argv.slice(2);

if (!command) {
  process.exit(usage());
}

switch (command) {
  case 'attach':
  case 'run':
    if (!session) {
      process.exit(usage());
    }
    process.exit(runZmx([command, session, ...rest]));
    break;
  case 'list':
    process.exit(runZmx(['list']));
    break;
  case 'history':
  case 'kill':
    if (!session) {
      process.exit(usage());
    }
    process.exit(runZmx([command, session]));
    break;
  case 'detach':
  case 'version':
    process.exit(runZmx([command]));
    break;
  default:
    process.exit(usage());
}

File: .opencode/skills/zmx-long-lived-shells/scripts/zmx_session.py
#!/usr/bin/env python3
'''Minimal wrapper around zmx for long-lived sessions.'''

import argparse
import subprocess
import sys


def run_zmx(args):
    cmd = ['zmx'] + args
    try:
        return subprocess.call(cmd)
    except FileNotFoundError:
        print('zmx not found. Install with: brew install neurosnap/tap/zmx', file=sys.stderr)
        return 127


def parse_args():
    parser = argparse.ArgumentParser(description='Wrapper for zmx session commands.')
    subparsers = parser.add_subparsers(dest='command', required=True)

    attach = subparsers.add_parser('attach', help='Attach to a session')
    attach.add_argument('session')
    attach.add_argument('cmd', nargs=argparse.REMAINDER, help='Command to run in session')

    run = subparsers.add_parser('run', help='Run a command in session without attaching')
    run.add_argument('session')
    run.add_argument('cmd', nargs=argparse.REMAINDER, help='Command to run in session')

    subparsers.add_parser('list', help='List sessions')

    history = subparsers.add_parser('history', help='Show session scrollback')
    history.add_argument('session')

    kill = subparsers.add_parser('kill', help='Kill a session')
    kill.add_argument('session')

    subparsers.add_parser('detach', help='Detach all clients from current session')
    subparsers.add_parser('version', help='Show zmx version')

    return parser.parse_args()


def main():
    args = parse_args()
    if args.command in ('attach', 'run'):
        return run_zmx([args.command, args.session] + args.cmd)
    if args.command == 'list':
        return run_zmx(['list'])
    if args.command == 'history':
        return run_zmx(['history', args.session])
    if args.command == 'kill':
        return run_zmx(['kill', args.session])
    if args.command == 'detach':
        return run_zmx(['detach'])
    if args.command == 'version':
        return run_zmx(['version'])
    return 1


if __name__ == '__main__':
    raise SystemExit(main())
